cli/src/template_builder.rs

```diff
@@ -888,10 +938,10 @@ fn builtin_string_methods<'a, L: TemplateLanguage<'a> + ?Sized>(
         |language, diagnostics, build_ctx, self_property, function| {
             let [needle_node] = function.expect_exact_arguments()?;
             // TODO: or .try_into_string() to disable implicit type cast?
-            let needle_property =
-                expect_stringify_expression(language, diagnostics, build_ctx, needle_node)?;
-            let out_property = (self_property, needle_property)
-                .map(|(haystack, needle)| haystack.contains(&needle));
+            let needle_pattern =
+                expect_string_pattern(language, diagnostics, build_ctx, needle_node)?;
+            let out_property =
+                (self_property, needle_pattern).map(|(haystack, needle)| needle.matches(&haystack));
```

Comment:
Since `contains()` implies substring matching, the current `StringMatcher` API wouldn't be useful here.

Maybe we can add `.matches(pattern)`, etc. instead? Since we don't have to support dynamic strings, the implementation will become simpler.